platform-metrics/src/main/java/org/hypertrace/core/serviceframework/metrics/PlatformMetricsRegistry.java

```diff
@@ -497,6 +488,20 @@ public static synchronized void stop() {
     isInit = false;
   }
 
+  static MultiGauge registerMultiGauge(final String gaugeName) {
```

Comment:
I would still suggest flipping this around to match the other meters. The Registry is what external code interacts with today, which returns a meter that is then used to report.

```
  public static ResizableGauge registerResizableGauge(
      String name // consider adding tags so we don't have to break the API later)
```

That would mean you'd remove the registration from the constructor of ResizableGauge (and make it package private). 

If you want to switch that around (since I think constructing a meter and registering it separately is a better design, also what micrometer does), we could do so but I would still keep the registration out of the ResizableGauge constructor and I would make this function in platform metrics registry generic (e.g. `<T extends Registrable> T register(T registrable)`)

Hope that makes sense, happy to discuss if not.

Reply:
Makes sense. Picking the first approach here for consistency.
> consider adding tags so we don't have to break the API later)

We may not have the labels/tags upfront while registering the gauge. We only get them from the query results. Did you mean taking in any additional Tags and passing down empty Tags from `monitorCustomMetrics`?

Reply:
> We may not have the labels/tags upfront while registering the gauge. We only get them from the query results. Did you mean taking in any additional Tags and passing down empty Tags from monitorCustomMetrics?

Yeah, I think the underlying multigauge allows both tags up front (presumably static things, like what service this is) and tags on the individual measurement reports. I suppose we could always overload it later if we don't need it now.